paint.sh

```diff
@@ -15,5 +15,5 @@ do
 		export GIT_AUTHOR_DATE="$d 12:$m:$s"
 		git commit --date="$d 12:$m:$s" -m "$i on $d" --no-gpg-sign --allow-empty
 	done
-done < ${0%/*}/dates.txt
+done < "$(dirname $0)/dates.txt"
```

Comment:
$0 should still even be quoted in that subshell. I know, it's nitpicking, but … ;-)

Reply:
escaped double quote or can I use a single quote or what?

Reply:
As you need the expansion of `$0` you need double quotes, like this:
```bash
done < "$(dirname "$0")/dates.txt"
```

NB: `"$(dirname "$0")"/dates.txt` and `"$(dirname "$0")"'/dates.txt'` would equally be valid and correct.

Reply:
ah, wasn't sure if bash was smart enough to handle the nested quoting properly. it's nice when things Just Work :stuck_out_tongue: